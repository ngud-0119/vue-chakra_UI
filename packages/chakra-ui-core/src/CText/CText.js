import CBox from '../CBox'
import { forwardProps } from '../utils'
import { baseProps } from '../config/props'

export default {
  name: 'CText',
  inject: ['$chakraTheme', '$chakraColorMode'],
  props: {
    as: {
      type: [String, Array],
      default: 'p'
    },
    ...baseProps
  },
  render (h) {
    return h(CBox, {
      props: {
        as: this.as,
        fontFamily: this.as === 'kbd' ? 'mono' : 'body',
        ...forwardProps(this.$props)
      }
    }, this.$slots.default)
  }
}
