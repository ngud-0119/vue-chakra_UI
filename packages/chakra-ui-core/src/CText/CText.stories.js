import { storiesOf } from '@storybook/vue'
import { CText } from '..'

storiesOf('UI | Text', module)
  .add('Paragraph', () => ({
    components: { CText },
    template: `
      <div>
        <CText>Text Component</CText>
      </div>
    `
  }))
  .add('As underline', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="u">Underline</CText>
      </div>
    `
  }))
  .add('As abbreviation', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="abbr">I18N</CText>
      </div>
    `
  }))
  .add('As citation', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="cite">Citation</CText>
      </div>
    `
  }))
  .add('As deleted', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="del">Deleted</CText>
      </div>
    `
  }))
  .add('With emphasis', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="em">Emphasis</CText>
      </div>
    `
  }))
  .add('As inserted', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="ins">Inserted</CText>
      </div>
    `
  }))
  .add('As keyboard', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="kbd">Ctrl + C</CText>
      </div>
    `
  }))
  .add('As highlighted', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="mark">Highlighted</CText>
      </div>
    `
  }))
  .add('As strikethrough', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="s">Strikethrough</CText>
      </div>
    `
  }))
  .add('As subscript', () => ({
    components: { CText },
    template: `
      <div>
        <CText d="inline-block">2</CText>
        <CText as="sub">sub</CText>
      </div>
    `
  }))
  .add('As superscript', () => ({
    components: { CText },
    template: `
      <div>
        <CText d="inline-block">5</CText>
        <CText as="sup">sup</CText>
      </div>
    `
  }))
  .add('isTruncated', () => ({
    components: { CText },
    template: `
      <div style="width: 80vw">
        <CText isTruncated>Extra super duper long naruto heading in the village hidden in the leaves by the first Hokage who used to be friedns with Madara Uchiha but they went their separate ways because they could find a common understing around the definition of peace in the ninja world.t</CText>
      </div>
    `
  }))
  .add('with noOfLines', () => ({
    components: { CText },
    template: `
      <div style="width: 80vw">
        <div>This will not show more than 3 lines</div>
        <CText noOfLines="3" mb="2" borderWidth="1px" borderColor="red.200">
          Extra super duper long naruto heading in the village hidden in the leaves by the first Hokage who used to be friedns with Madara Uchiha but they went their separate ways because they could find a common understing around the definition of peace in the ninja world. Add text to make it long enough
        </CText>
      </div>
    `
  }))
  .add('with textDecor', () => ({
    components: { CText },
    template: `
      <div style="width: 80vw; padding-top: 20px;">
        <div>Some decorations:</div>
        <CText p="4" mb="2" borderWidth="1px" borderColor="red.200" textDecor="overline">
          Extra super duper long naruto heading in the village hidden in the leaves by the first Hokage who used to be friedns with Madara Uchiha but they went their separate ways because they could find a common understing around the definition of peace in the ninja world.
        </CText>
        <CText p="4" mb="2" borderWidth="1px" borderColor="blue.200" textDecor="line-through">
          Extra super duper long naruto heading in the village hidden in the leaves by the first Hokage who used to be friedns with Madara Uchiha but they went their separate ways because they could find a common understing around the definition of peace in the ninja world.
        </CText>
        <CText p="4" mb="2" borderWidth="1px" borderColor="green.200" textDecor="underline">
          Extra super duper long naruto heading in the village hidden in the leaves by the first Hokage who used to be friedns with Madara Uchiha but they went their separate ways because they could find a common understing around the definition of peace in the ninja world.
        </CText>
      </div>
    `
  }))
