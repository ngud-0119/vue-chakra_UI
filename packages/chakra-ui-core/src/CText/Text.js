import CBox from '../Box'
import { forwardProps } from '../utils'
import { baseProps } from '../config/props'

export default {
  name: 'CText',
  inject: ['$theme', '$colorMode'],
  props: {
    as: {
      type: [String, Array],
      default: 'p'
    },
    ...baseProps
  },
  render (h) {
    return h(CBox, {
      props: {
        as: this.as,
        fontFamily: this.as === 'kbd' ? 'mono' : 'body',
        ...forwardProps(this.$props)
      }
    }, this.$slots.default)
  }
}
