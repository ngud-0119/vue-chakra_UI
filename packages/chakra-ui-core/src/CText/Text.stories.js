import { storiesOf } from '@storybook/vue'
import { CText } from '..'

storiesOf('UI | Text', module)
  .add('Paragraph', () => ({
    components: { CText },
    template: `
      <div>
        <CText>Text Component</CText>
      </div>
    `
  }))
  .add('As underline', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="u">Underline</CText>
      </div>
    `
  }))
  .add('As abbreviation', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="abbr">I18N</CText>
      </div>
    `
  }))
  .add('As citation', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="cite">Citation</CText>
      </div>
    `
  }))
  .add('As deleted', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="del">Deleted</CText>
      </div>
    `
  }))
  .add('With emphasis', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="em">Emphasis</CText>
      </div>
    `
  }))
  .add('As inserted', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="ins">Inserted</CText>
      </div>
    `
  }))
  .add('As keyboard', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="kbd">Ctrl + C</CText>
      </div>
    `
  }))
  .add('As highlighted', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="mark">Highlighted</CText>
      </div>
    `
  }))
  .add('As strikethrough', () => ({
    components: { CText },
    template: `
      <div>
        <CText as="s">Strikethrough</CText>
      </div>
    `
  }))
  .add('As subscript', () => ({
    components: { CText },
    template: `
      <div>
        <CText d="inline-block">2</CText>
        <CText as="sub">sub</CText>
      </div>
    `
  }))
  .add('As superscript', () => ({
    components: { CText },
    template: `
      <div>
        <CText d="inline-block">5</CText>
        <CText as="sup">sup</CText>
      </div>
    `
  }))
