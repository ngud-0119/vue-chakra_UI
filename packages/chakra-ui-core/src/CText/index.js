import CText from './Text'
export default CText
