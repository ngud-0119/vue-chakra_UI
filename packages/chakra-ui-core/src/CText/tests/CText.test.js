import CText from '..'
import { render } from '@/tests/test-utils'

const renderComponent = (props) => {
  const inlineAttrs = (props && props.inlineAttrs) || ''
  const base = {
    components: { CText },
    template: `<CText data-testid="text" ${inlineAttrs}>Text Works</CText>`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()

  expect(asFragment()).toMatchSnapshot()
})

it('should change the style', () => {
  const inlineAttrs = 'd="flex"'
  const { asFragment, getByTestId } = renderComponent({ inlineAttrs })

  const text = getByTestId('text')

  expect(asFragment()).toMatchSnapshot()
  expect(text).toHaveStyle('display: flex')
})

it.each`
  as
  ${'u'}
  ${'abbr'}
  ${'cite'}
  ${'del'}
  ${'em'}
  ${'ins'}
  ${'kbd'}
  ${'mark'}
  ${'s'}
  ${'sub'}
  ${'sup'}}
`(
  'should display text type as $as',
  ({ as }) => {
    const inlineAttrs = `as=${as}`
    const { asFragment } = renderComponent({ inlineAttrs })

    expect(asFragment()).toMatchSnapshot()
  }
)
