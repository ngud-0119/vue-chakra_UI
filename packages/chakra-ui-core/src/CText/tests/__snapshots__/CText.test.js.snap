// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change the style 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--chakra-fonts-body);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<p
    class="emotion-0"
    data-chakra-component="CText"
    data-testid="text"
  >
    Text Works
  </p>
</DocumentFragment>
`;

exports[`should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--chakra-fonts-body);
}

<p
    class="emotion-0"
    data-chakra-component="CText"
    data-testid="text"
  >
    Text Works
  </p>
</DocumentFragment>
`;
